openmdao/core/system.py

```diff
@@ -306,7 +306,7 @@ def _set_vars_as_remote(self):
             meta['remote'] = True
 
     def fd_jacobian(self, params, unknowns, resids, total_derivs=False,
```

Comment:
I just noticed a preexisting issue in fd_jacobian.  It looks like we don't reset the value of param_src, so we could end up using a stale value from an earlier iteration of the loop, which looks like it could end up giving us the wrong indices down below.